uktrade/data-hub-frontend
# Add countries when adding an interaction - Part 1

## Description of change

**Hidden behind a feature flag called "interaction-add-countries"** so this PR should have no changes, unless this feature flag is present - this is to help make the PRs for this feature smaller.

To capture more up to date information about future countries of interest, countries currently exporting to or countries a company is not interested in exporting to, this adds a new question when adding an interaction for Export and Other "themes" to ask if any countries were discussed. If countries were discussed then three additional fields are revealed to ask for a list of countries for 

- Future countries of interest
- Countries currently exporting to
- Countries not interested in

Each of these fields is a Typeahead and there were a few modifications required to make the typeahead work in this scenario.

## Test instructions

Add a new interaction with a theme/type of "Export" or "Other" with any sub type selected, you should see a new question at the bottom asking "Were any countries discussed?"

_What should I see?_
 
## Screenshots
### Before

<img width="736" alt="Screenshot 2019-11-28 at 19 18 24" src="https://user-images.githubusercontent.com/1481883/69828486-f4440b00-1213-11ea-861c-41ed55ad502b.png">


### After 

<img width="723" alt="Screenshot 2019-11-28 at 19 18 34" src="https://user-images.githubusercontent.com/1481883/69828496-fad28280-1213-11ea-9375-b1961e2c546a.png">


## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied. 
https://github.com/uktrade/data-hub-frontend/blob/develop/docs/Code%20review%20guidelines.md"
